datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -221,13 +223,14 @@ int getTasksSize() {
     return _tasks.size();
   }
 
-  private Producer<K, V> initializeProducer(DatastreamTask task) throws InterruptedException {
+  private Producer<K, V> initializeProducer() throws InterruptedException {
     // Must be protected by a lock to avoid creating duplicate producers when multiple concurrent
     // sends are in-flight and _kafkaProducer has been set to null as a result of previous
     // producer exception.
     try {
+      System.out.println("wait for lock");
       _producerLock.lock();
-
+      System.out.println("got lock");
```

Comment:
Did you mean to remove these System.out.println()?

Reply:
done